### DISCUSSION TITLE
TextInput grouping with rightSection Button or right Select or any other general input.

### DISCUSSION THREAD

**Discussion by najibghadri** [mantinedev/mantine#5713] [Q&A] [answered]
Hello! How can we do these in Mantine:
<img width="491" alt="Screenshot 2024-02-03 at 15 01 39" src="https://github.com/mantinedev/mantine/assets/11639734/fde3ab19-dfa8-4fa8-bb05-dc4ab6223930">
<img width="407" alt="Screenshot 2024-02-03 at 15 02 48" src="https://github.com/mantinedev/mantine/assets/11639734/ce4a6584-ca82-4824-8c19-166be311e9ca">

I tried these with Mantine but neither worked. If I set rightSectionWidth with button that kind of works, but it's hacky, if there is translation in our website it won't work for other text lenghts...
Tried with Button.Group and rightSections:
```
        <TextInput placeholder="Your name" label="Full name" withAsterisk />
        <TextInput
          placeholder="Your name"
          label="Full name"
          withAsterisk
          rightSection={
            <Select
              placeholder="Pick one"
              data={[
                { value: 'react', label: 'React' },
                { value: 'ng', label: 'Angular' },
                { value: 'svelte', label: 'Svelte' },
                { value: 'vue', label: 'Vue' },
              ]}
              styles={{
                input: {},
              }}
            />
          }
        />
        <Button.Group>
          <TextInput placeholder="Your name" label="Full name" withAsterisk />
          <Button variant="filled">Submit</Button>
        </Button.Group>
        <TextInput
          placeholder="Your name"
          label="Full name"
          withAsterisk
          rightSectionWidth={140}
          rightSection={
            <Button w={140} radius="md">
              Choose file
            </Button>
          }
        />
```

<img width="725" alt="Screenshot 2024-02-03 at 15 07 16" src="https://github.com/mantinedev/mantine/assets/11639734/e9ef83dc-3625-4448-9ace-fa356baf2769">

I think these are pretty common functionalities for a UI lib, I think there should be a way to do these without hacky pixel work, because that isn't a maintainable solution. If there is no way to do this nicely in Mantine then at least try to give us some hints if and how we could implement a "Group" component that can fix this generally to any input like these (like Button.Group), or if you think that's not a viable way please you could warn in advance. Thanks!

**Comment by rtivital** [comment, marked as answer]
Mantine inputs do not support this feature. The only option is to set rightSectionWidth property.

**Reply by najibghadri** [reply]
Thanks!
